courses/models.py

```diff
@@ -33,6 +37,30 @@ class Course(models.Model):
     def __str__(self):
         return self.title
 
+    class Meta:
+        unique_together = ('program', 'position_in_program',)
+
+    def get_next_run(self):
+        """
+        Gets the next run associated with this course.
+        Note that it could be current as long as the enrollment window
+        has not closed.
+        """
+        now = datetime.now(pytz.utc)
+        next_run_set = self.courserun_set.filter(
+            # there is only the start date
+            (models.Q(enrollment_start=None) & models.Q(enrollment_end=None) &
+             models.Q(end_date=None) & models.Q(start_date__lte=now)) |
+            # there is no enrollment date but the course has not ended yet
+            (models.Q(enrollment_end=None) & models.Q(end_date__gte=now)) |
```

Comment:
this case is covered by the following expression (unless I misunderstood what you mean)